server/base/src/main/java/org/apache/accumulo/server/util/ServiceStatusCmd.java

```diff
@@ -0,0 +1,336 @@
+    } else {
+      StringBuilder sb = new StringBuilder(8192);
+      report.report(sb);
+      System.out.println(sb);
+    }
+  }
+
+  /**
+   * The manager paths in ZooKeeper are: {@code /accumulo/[IID]/managers/lock/zlock#[NUM]} with the
+   * lock data providing host:port.
+   */
+  @VisibleForTesting
+  StatusSummary getManagerStatus(final ZooReader zooReader, String zRootPath) {
+    String lockPath = zRootPath + Constants.ZMANAGER_LOCK;
+    return getStatusSummary(ServiceStatusReport.ReportKey.MANAGER, zooReader, lockPath);
+  }
+
+  /**
+   * The monitor paths in ZooKeeper are: {@code /accumulo/[IID]/monitor/lock/zlock#[NUM]} with the
+   * lock data providing host:port.
+   */
+  @VisibleForTesting
+  StatusSummary getMonitorStatus(final ZooReader zooReader, String zRootPath) {
+    String lockPath = zRootPath + Constants.ZMONITOR_LOCK;
+    return getStatusSummary(ServiceStatusReport.ReportKey.MONITOR, zooReader, lockPath);
+  }
+
+  /**
+   * The tserver paths in ZooKeeper are: {@code /accumulo/[IID]/tservers/[host:port]/zlock#[NUM]}
+   * with the lock data providing TSERV_CLIENT=host:port.
+   */
+  @VisibleForTesting
+  StatusSummary getTServerStatus(final ZooReader zooReader, String zRootPath) {
+    String lockPath = zRootPath + Constants.ZTSERVERS;
+    return getServerHostStatus(zooReader, lockPath, ServiceStatusReport.ReportKey.T_SERVER);
+  }
+
+  /**
+   * The sserver paths in ZooKeeper are: {@code /accumulo/[IID]/sservers/[host:port]/zlock#[NUM]}
+   * with the lock data providing [UUID],[GROUP]
+   */
+  @VisibleForTesting
+  StatusSummary getScanServerStatus(final ZooReader zooReader, String zRootPath) {
+    String lockPath = zRootPath + Constants.ZSSERVERS;
+    return getServerHostStatus(zooReader, lockPath, ServiceStatusReport.ReportKey.S_SERVER);
+  }
+
+  /**
+   * handles paths for tservers and servers with the lock stored beneath the host: port like:
+   * {@code /accumulo/IID/[tservers | sservers]/HOST:PORT/[LOCK]}
+   */
+  private StatusSummary getServerHostStatus(final ZooReader zooReader, String basePath,
+      ServiceStatusReport.ReportKey displayNames) {
+    AtomicInteger errorSum = new AtomicInteger(0);
+
+    Set<String> hostNames = new TreeSet<>();
+    Set<String> groupNames = new TreeSet<>();
+
+    var nodeNames = readNodeNames(zooReader, basePath);
+
+    nodeNames.getSecond().forEach(name -> {
+      var lock = readNodeNames(zooReader, basePath + "/" + name);
+      lock.getSecond().forEach(l -> {
+        var r = readNodeData(zooReader, basePath + "/" + name + "/" + l);
+        int err = r.getFirst();
+        if (err > 0) {
+          errorSum.addAndGet(r.getFirst());
+        } else {
+          // process resource groups
+          var payload = r.getSecond();
+          String[] tokens = payload.split(",");
+          String groupSeparator = "";
+          if (tokens.length == 2) {
+            groupNames.add(tokens[1]);
+            groupSeparator = tokens[1] + ": ";
+          }
```

Comment:
Is this parsing the lock content represented by the ServerServices object?  If so, be aware that this changed in 3.1, ref: https://github.com/apache/accumulo/blob/main/server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java#L664